src/cassandra/custom_payload.rs

```diff
@@ -1,7 +1,56 @@
-// use cassandra_sys::cass_custom_payload_free;
-// use cassandra_sys::cass_custom_payload_new;
-// use cassandra_sys::cass_custom_payload_set;
+use crate::cassandra::error::*;
+use crate::cassandra::util::Protected;
 
-// use cassandra_sys::CassCustomPayload as _CassCustomPayload;
-//
-// pub struct CustomPayload(_CasCustomPayload);
+use crate::cassandra_sys::cass_custom_payload_free;
+use crate::cassandra_sys::cass_custom_payload_new;
+use crate::cassandra_sys::cass_custom_payload_set;
+use crate::cassandra_sys::cass_custom_payload_set_n;
+use crate::cassandra_sys::CassCustomPayload as _CassCustomPayload;
+use std::collections::HashMap;
+use std::os::raw::c_char;
+
+pub type CustomPayloadResponse = HashMap<String, Vec<u8>>;
+
+/// Custom payloads not fully supported yet
+#[derive(Debug)]
+pub struct CustomPayload(*mut _CassCustomPayload);
+
```

Comment:
I think you can probably make this `Send`, though obviously not `Sync` - see justification elsewhere e.g. https://github.com/Metaswitch/cassandra-rs/blob/master/src/cassandra/collection.rs#L432 .

Reply:
I just moved the code from batch.rs; do you want me to do this?

Reply:
Yes please. If we're going to start using these, we're going to want them to play nicely with futures-0.3.

Reply:
Fixed.